src/native/http_connection_manager.c

```diff
@@ -266,74 +283,100 @@ JNIEXPORT jlong JNICALL Java_software_amazon_awssdk_crt_http_HttpClientConnectio
 cleanup:
     aws_jni_byte_cursor_from_jbyteArray_release(env, jni_endpoint, endpoint);
 
-    return (jlong)conn_manager;
+    if (binding->manager == NULL) {
+        s_destroy_manager_binding(binding);
```

Comment:
Exception is probably better given that the error information is otherwise lost and all you get is a generic IllegalStateException.  To start off, fixed the non-null return on failure.  Will look into exception next.